docs/EN_US/ContainerizedHPCC/ContainerizedMods/TroubleshootingHelmDeployments.xml

```diff
@@ -0,0 +1,516 @@
+    effective Kubernetes administrator and troubleshoot your deployments with
+    confidence.</para>
+  </sect1>
+
+  <sect1 id="check-the-status-of-services" role="brk">
+    <title>Check the Status of Services</title>
+
+    <para>Services expose applications running on a cluster.</para>
+
+    <para>The <emphasis>kubectl get services</emphasis> command is a powerful
+    tool for troubleshooting Kubernetes deployments. It provides a concise
+    overview of the services running in your cluster, helping you identify
+    potential issues and their root causes.</para>
+
+    <para>To get a quick overview of services status, use the following
+    command in a terminal window:</para>
+
+    <para><programlisting>kubectl get services</programlisting></para>
+
+    <para>This lists all services in your cluster, along with their type,
+    internal and external IP addresses, port, and uptime (Age).</para>
+
+    <para>If you have deployments to more than one namespace, use this
+    command:</para>
+
+    <para><programlisting>kubectl get service -A</programlisting></para>
+
+    <para>This lists all services in all namespaces.</para>
+
+    <para>If a service that should have an external IP listed does not have
+    one displayed, that pod has an issue.</para>
+  </sect1>
+
+  <sect1 id="describe-a-service">
+    <title>Describe a Service</title>
+
+    <para>Use the <emphasis role="strong">kubectl</emphasis> command-line tool
+    to get detailed information about a service. By describing a service in
+    Kubernetes, you can gain valuable insights into its configuration, health,
+    and how it interacts with pods.</para>
+
+    <para>To get detailed information about a service, use the following
+    command in a terminal window:</para>
+
+    <para><programlisting>kubectl describe service &lt;service-name&gt;</programlisting></para>
+
+    <para>The output provides detailed information about the service,
+    including the service's IP address, port, selectors, and other
+    details.</para>
+
+    <para>If you have deployments to more than one namespace, use this
+    command:</para>
+
+    <para><programlisting>kubectl describe service &lt;service-name&gt;  -A</programlisting></para>
+
+    <para>This describes the service across all namespaces.</para>
+  </sect1>
+
+  <sect1 id="ViewingPodLogs">
+    <title>Viewing Pod Logs</title>
+
+    <para>Viewing pod logs is a crucial step in troubleshooting Helm
+    deployments because it provides real-time insights into the behavior and
+    errors occurring within your application containers. By analyzing these
+    logs, you can quickly identify and address a wide range of issues.</para>
+
+    <para>To view the logs of a specific pod, use the following command in a
+    terminal window:</para>
+
+    <para><programlisting>kubectl logs &lt;pod-name&gt;</programlisting></para>
+
+    <para>This returns the entire log for a pod.</para>
```

Comment:
how much is returned will depend on the log rotation policy, which I think will depend on the container runtime retention policy.
Just saying, if the pod's logs is GB's long - kubectl logs is unlikely to return it all..

Reply:
Might also be worth noting something about pods with multiple containers. Dali by default has 2 (dali itself and sasha-coalescer). By default kubectl logs <podname> will return the logs of the 1st container in the pod.
If you need the logs for a specific container you need to say:
kubectl logs <podname> -c <containername>